src/Sarif/FileRegionsCache.cs

```diff
@@ -274,23 +275,48 @@ private void PopulateCharLength(NewLineIndex lineIndex, Region region)
             {
                 region.CharLength = charLength;
             }
-            SarifUtilities.DebugAssert(region.CharLength == charLength);
+            Assert(region.CharLength == charLength);
         }
 
         private NewLineIndex GetNewLineIndex(Uri uri, out string fileText)
         {
-            fileText = null;
+            Tuple<string, NewLineIndex> entry = _cache[uri.LocalPath];
+         
+            fileText = entry.Item1;
+            return entry.Item2;
+        }
+
+        private Tuple<string, NewLineIndex> BuildIndexForFile(string localPath)
+        {
+            string fileText = null;
+            NewLineIndex index = null;
 
             // We will expand this code later to construct all possible URLs from
             // the log file, bearing in mind things like uriBaseIds. Also, we could
             // consider downloading and caching web-hosted source files.
             try
             {
-                fileText = _fileSystem.ReadAllText(uri.LocalPath);
+                fileText = _fileSystem.ReadAllText(localPath);
             }
             catch (IOException) { }
 
-            return fileText != null ? new NewLineIndex(fileText) : null;
+            if (fileText != null)
+            {
+                index = new NewLineIndex(fileText);
+            }
+
+            return new Tuple<string, NewLineIndex>(fileText, index);
+        }
+
+        private static void Assert(bool condition)
+        {
+            // Placeholder to report issues in a situationally appropriate way.
+            //  We don't want Multitool rewrite to blow up.
+            //  We don't want unit tests for invalid Regions to block on asserts; we want to verify the code leaves those results alone.
+            //  We may want console or output log output when invalid Regions are detected during Multitool rewrite use.
```

Comment:
>use [](start = 116, length = 3)

Handling asserts in a situationally appropriate way... nice idea, I wonder that I've never seen it before. Perhaps you can inject an assert-handler function into this class:

```
private readonly Action<bool> _assertHandler;

FileRegionCache(... , Action<bool> assertHandler)
{
    ... 
    _assertHandler = assertHandler ?? ((bool condition) => Debug.Assert(condition));
}

...

private static void Assert(bool condition)
{
    _assertHandler(condition);
}
```

Reply:
Yes. I'm not sure where we want to land, so I just added the stub and redirected all of the Asserts so we didn't lose that verification logic.